lib/charms/opensearch/v0/opensearch_relation_provider.py

```diff
@@ -343,8 +354,9 @@ def _on_relation_broken(self, event: RelationBrokenEvent) -> None:
         if not self.unit.is_leader():
             return
         if self._unit_departing(event.relation):
-            # This unit is being removed, so don't update the relation.
+            # This unit is being removed.
             self.charm.peers_data.delete(Scope.UNIT, self._depart_flag(event.relation))
+            self.update_endpoints(event.relation, omit_endpoints={self.charm.unit_ip})
```

Comment:
can you explain the logic here?

Reply:
Effectively, we want to update the endpoints but we want to exclude the IP of the departing unit so the client app won't expect that unit to be reachable. I'm now doing this in the departing hook, using `event.departing_unit`'s IP.